engine/src/main/java/org/hibernate/validator/internal/engine/ConfigurationImpl.java

```diff
@@ -483,16 +494,23 @@ private void applyXmlSettings(ValidationBootstrapParameters xmlParameters) {
 		}
 	}
 
-	private boolean isJavaFxInClasspath() {
-		return isClassPresent( "javafx.application.Application" );
-	}
-
-	private boolean isClassPresent(String className) {
+	private boolean isJavaFxInClasspath(final ClassLoader classLoader) {
 		try {
-			run( LoadClass.action( className, getClass().getClassLoader() ) );
```

Comment:
I think you can keep the current `LoadClass` (see comment above), but if you really need a new version which for example does not check the TCCL, then you should either modify `LoadClass` or create a new `LoadClassXYZ`.